core/field.ts

```diff
@@ -1263,6 +1264,16 @@ export abstract class Field<T = any>
     return false;
   }
 
+  /**
+   * Get the variable model if this field refers to one.
+   *
+   * @returns The field's variable, or null if none exists.
+   * @internal
+   */
+  getVariable(): VariableModel | null {
+    return null;
```

Comment:
So as the code currently exists, this is a breaking change because we're requiring folks to implement the `getVariable()` method to return the variable they're referencing, which we didn't require before.

We can make this backwards compatible by having `getVariable` get the variable based on the value if `referencesVariables` is true.

Please add a unit test for this as well using a mock `Field` subclass.

Reply:
Thanks. I suppose we could do something like the following, and deprecate this check if/when `referencesVariables` is dropped. I'm not entirely sure where to get the variable from, however. It might not live on the source block's workspace.

Or maybe we could handle backwards compatibility at the `getVars` stage where it was calling `getValue()`. Most other places in the API already seem to be correctly using `getVariable` already. I'll have a look.

```js
//  field.ts   //

  /**
   * Get the variable models this field refers to, if any.
   * 
   * @returns The field's variable, or null if none exists.
   */
  getVariables(): VariableModel[] {
    // Previously, when referencesVariables was true, we assumed getValue()
    // returned a variable ID. Keep allowing this for backwards compatiblity.
    if (this.referencesVariables()) {
      const variable = this.getSourceBlock()?.workspace.getVariableById(
        this.getValue() as string,
      )
      return variable ? [variable] : [];
    }

    // Fields don't reference variables by default. Subclasses are expected
    // to override this method if they do.
    return [];
  }
```

Reply:
Addressed, see follow-up notes in main PR conversation thread.